appservice/src/tree/DeploymentsTreeItem.ts

```diff
@@ -92,10 +93,27 @@ export class DeploymentsTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
 
     public async refreshImpl(): Promise<void> {
         const siteConfig: SiteConfig = await this.root.client.getSiteConfig();
+        const sourceControl: SiteSourceControl = await this.root.client.getSourceControl();
         if (siteConfig.scmType === ScmType.GitHub || siteConfig.scmType === ScmType.LocalGit) {
             this.contextValue = DeploymentsTreeItem.contextValueConnected;
         } else {
             this.contextValue = DeploymentsTreeItem.contextValueUnconnected;
         }
+
+        this.description = this.getDescriptionFromScmType(siteConfig, sourceControl);
+    }
+
+    private getDescriptionFromScmType(siteConfig: SiteConfig, sourceControl: SiteSourceControl): string {
```

Comment:
I would prefer to use getters for `description` and `contextValue`. Something like this where you reference private variables like `_scmType` and `_repoUrl` that are updated in `refreshImpl`:
```typescript
private _scmType: string;
private _repoUrl?: string;

public get description(): string {
    switch (this._scmType) {
        case ScmType.LocalGit:
            return localize('localGit', 'Local Git');
        case ScmType.GitHub:
            // remove github from the repoUrl which leaves only the org/repo names
            return this._repoUrl ? this._repoUrl.substring('https://github.com/'.length) : localize('github', 'Github');
        case ScmType.None:
        default:
            return '';
    }
}
```